### PR TITLE
Backend rework for better profile handling

### PR DESCRIPTION
**Pull request by Anko59**
- Form message to fill to create new users
- Enable creation of `MistralAgent` without a profile to allow for profile creation
- API endpoint to handle `POST` requests different from the `chat_view` endpoint
- Rewrite of `views.py` and `main.js` to handle new endpoint
- `models.py` file to store the `UserProfile` object and functions to serialize and unserlaize it
- Storage of `UserProfile` objects handled by django by default in prostgres table
- New user management required a rewrite of `CVEditor`
- `/media/` directory to handle uploaded user files
- `nginx` server to serve media files at `aiden.dev.localhost/media/`
Ready to merge

### CODE REVIEW

File: aiden_app/agents/agent.py [hunk 1]
```diff
@@ -1,58 +1,101 @@
...
-            message_class(role="system", content=SYSTEM_PROMPT),
-            message_class(role="system", content=json.dumps(profile)),
-            message_class(role="assistant", content=START_CHAT_PROMPT),
-        ]
+    def __init__(self, message_class: Any):
+        self.tool_aggregator = ToolAggregator([])
         self.message_class = message_class
-        self.messages: List[Dict[str, Any]] = start_messages
+        self.messages: List[Dict[str, Any]] = []
+        self.profile: Dict[str, str] = {}
```
Comment (Smirkey) [inline aiden_app/agents/agent.py]: we can use directly `list` and `dict` with our python version but that"s not very important
```suggestion
        self.messages: list[dict[str, Any]] = []
        self.profile: dict[str, str] = {}
```

File: aiden_app/agents/agent.py [hunk 2]
```diff
@@ -1,58 +1,101 @@
...
+        ]
+        self.messages = start_messages
+        return self
+
+    def create_profile(self, profile_info: dict[str, str]):
+        user_data_dir = os.path.join(os.path.dirname(__file__), "../user_data/")
+        profile_schema_path = os.path.join(user_data_dir, "cv_schema.json")
+        profile_schema = json.load(open(profile_schema_path))
+        example_profile_path = os.path.join(user_data_dir, "john_doe/default_profile.json")
+        example_profile = json.load(open(example_profile_path))
```
Comment (Smirkey) [inline aiden_app/agents/agent.py]: for this I think using pathlib would be easier than os, but not very important aswell

```python
from pathlib import Path
import json

user_data_dir = Path(__file__).resolve().parent / "../user_data/"
profile_schema_path = user_data_dir / "cv_schema.json"
profile_schema = json.loads(profile_schema_path.read_text())
example_profile_path = user_data_dir / "john_doe" / "default_profile.json"
example_profile = json.loads(example_profile_path.read_text())
```

Comment (Smirkey) [reply aiden_app/agents/agent.py]: Also this user data dir should be a constant defined somewhere what do you think?
Aiden app should create this dir

File: aiden_app/views.py [hunk 3]
```diff
@@ -2,110 +2,115 @@
...
-        response = {
-            "role": "assistant",
-            "content": START_CHAT_PROMPT,
-            "is_last": False,
-            "documents": documents,
-            "tokens_used": 0,
-        }
-        return response, agent
+
+@api_view(["POST"])
```
Comment (Smirkey) [inline aiden_app/views.py:25]: Nice

**Review by Smirkey — COMMENTED** [review]
Some initial comments
Did not test locally yet

File: aiden_app/agents/agent.py (hunk 1 above)
Comment (Anko59) [reply aiden_app/agents/agent.py]: I agree, I like it more, it's just copilot generating some outdated typing

File: aiden_app/agents/agent.py (hunk 2 above)
Comment (Anko59) [reply aiden_app/agents/agent.py]: We should move to a Django way to store and retrive user data. I doubt that a user_data dir in the aiden_app folder is the best way to do it.

**Review by Smirkey — APPROVED** [review]
as seen yesterday, works like a charm !

**Pull request merged**